### PR TITLE
Fix Reboot Handler crash

### LINKED ISSUE #3682: Reboot Handler causes a crash
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

_No response_

### What Operating System are you using?

Linux x86_64

### If Other OS, please describe

Debian 12

### Relevant hardware info

_No response_

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Our reboot handler crashes due probably to undefined destruction order of global objects.  I had previously reported this on #3531 when I encountered this on the Windows 3.1 installer.  The problem is more generic than that though.  I think every time this reboot handler is called, it's a problem.

The easiest way to reproduce is to use the `boot` command to boot a non-bootable floppy image.  It will print an error of no system disk found.  Press enter and it will enter the reboot handler.

https://github.com/dosbox-staging/dosbox-staging/blob/5c0e311d4de8d2fc39b601a58a1212fa68c42777/src/ints/bios.cpp#L1013-L1031

With `machine = svga_s3` it triggers a SIGABRT from inside libc code due to heap corruption with the message "corrupted size vs. prev_size in fastbins"

With `machine = pcjr` you instead get a SIGSEGV when the Config destructor attempts to write to invalid memory.

This may even be different on different operating systems.  Above was tested on Debian 12.

### Steps to reproduce the behaviour.

Explain how to reproduce
1.  boot somefloppy.img
2. Press enter
3.  You should see a message from the reboot handler saying "Reboot Requested" followed by a crash after a few seconds

### Download URL of affected game or software

_No response_

### Your configuration

_No response_

### Provide a Log

_No response_

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

### PR DESCRIPTION
**Pull request by weirddan455**
# Description

Previously, this was doing `throw 1;` for some reason.  Not exactly sure what the end result of that was supposed to be.  I assume it would exit the program and throw an error since I don't know if we actually handle that exception anywhere.

Changed this to use the same restart function that is used by the restart hotkey as well as `config -r`.  This simply reloads DOSBox which is probably the result that we want here.

## Related issues

Fixes #3682

# Manual testing

Found a Space Quest 1 floppy image that is non-bootable.  Ran `boot sq1.img`.  This displayes a non-system disk error and, after pressing Enter, calls into the Restart Handler.  The Restart Handler now restarts DOSBox correctly instead of crashing.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by weirddan455** [comment]
I think GitHub must have updated their MSVC toolchain.  I was getting a build failure inside `cdrom_win32.cpp` (undefined macros).  Fixed by adding a header file but I can't see how that would have been caused by anything I did in this PR.

These rolling CI toolchains man :roll_eyes:   I guess MSVC is still better than MSYS2 in that our dependencies aren't rolling but they're randomly changing the header files or something.

File: src/ints/bios.cpp [hunk 1]
```diff
@@ -1026,7 +1026,7 @@ static Bitu Reboot_Handler(void) {
 	const auto start = PIC_FullIndex();
 	while ((PIC_FullIndex() - start) < 3000.0)
 		CALLBACK_Idle();
-	throw 1;
```
Comment (johnnovak) [inline src/ints/bios.cpp:1029]: Throwing exceptions for flow control... That's just ghastly 🥶 🤦🏻

**Review by johnnovak — APPROVED** [review]
Looks fine @weirddan455, just address @FeralChild64's comment please and then it's good to go 🚀

File: include/control.h [hunk 2]
```diff
@@ -159,4 +159,9 @@ class Config {
 	Verbosity GetStartupVerbosity() const;
 };
 
+using ConfigPtr = std::unique_ptr<Config>;
+extern ConfigPtr control;
+
+void restart_dosbox(std::vector<std::string> &parameters = control->startup_params);
```
[the remark this answers is not in the document]
Comment (weirddan455) [reply include/control.h:165]: Good call.  I wasn't aware of that keyboard reset function.  Added.

**Pull request merged**